File: jupyter2/problems/typedef/time/jl/time.jl
### begin my answer

### end my answer

function main()
    @assert abs(time_diff(Time(12, 0, 0.), Time(11, 30, 0.)) - 1800.) < 1e-6
    @assert abs(time_diff(Time(23, 59, 59.9), Time(0, 0, 0.)) - 86399.9) < 1e-6
    println("OK")
end

main()

File: jupyter2/problems/typedef/time/rs/time.rs
/** begin my answer */

/** end my answer */

fn main() {
    assert!((time_diff(Time {hour: 12, minute: 0, second: 0.0}, Time {hour: 11, minute: 30, second: 0.0}) - 1800.0).abs() < 1e-6);
    assert!((time_diff(Time {hour: 23, minute: 59, second: 59.9}, Time {hour: 0, minute: 0, second: 0.0}) - 86399.9).abs() < 1e-6);
    println!("OK")
}
